.gitignore

```diff
@@ -3,3 +3,7 @@ system/config/app.config.local.php
 
 *.log
 *.psd
+
+/nbproject/
+/vendor/
+composer.lock
```

Comment:
composer.lock should not be ignored. It is the file that contains the version downloaded and tested in DEV.  So that in PROD, you just have to `composer install` (and not `composer update`) to install only the tested version.